src/main/java/me/arminb/sara/dao/UserDAOImpl.java

```diff
@@ -0,0 +1,66 @@
+package me.arminb.sara.dao;
+
+import com.mongodb.BasicDBObject;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoCursor;
+import com.mongodb.client.MongoDatabase;
+import me.arminb.sara.entities.User;
+import org.bson.Document;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Repository("userDAO")
+public class UserDAOImpl implements UserDAO {
+
+    @Autowired
+    private MongoDatabase database;
+    //private MongoTemplate mongoTemplate;
+
+    @Override
+    public void findAll() {
+        try {
+            MongoCollection<Document> collection = database.getCollection("users");
+            for (Document doc : collection.find()) {
+                System.out.println(doc.toJson());
```

Comment:
this should return a list of users to be sent as a response to the REST request. You should not print anything here